app/src/models.py

```diff
@@ -48,4 +48,82 @@ def set_password(self, password):
+    QUIZ='Quiz'
+
+class Subject(Enum):
+    CALCULUS='Calculus'
+    LINALG='Linear Algebra'
+    BIOLOGY='Biology'
+    CHEMISTRY='Chemistry'
+    PHYSICS='Physics'
+    ENGLISH='English'
+    FRENCH='French'
+    COMSCI='Computer Science'
+    # ... add/take away as needed
+
+# Quizzes/Tasks
+class Activity(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    title = db.Column(db.String(100), nullable=False)
+    # establish relationship between 'quiz' and 'user' model
+    user_id = db.Column(db.Integer, ForeignKey('user.id'), nullable=False)
+    activity_type = db.Column(SQLAlchemyEnum(ActivityType), nullable=False, default=ActivityType.TASK)
+    subject_type = db.Column(SQLAlchemyEnum(Subject), nullable=False, default=Subject.BIOLOGY)
+    start_time = db.Column(db.DateTime, nullable=True, default=func.now()) 
+    end_time = db.Column(db.DateTime, nullable=True, default=func.now())
+    score = db.Column(db.Float, nullable=False)
+    content = db.Column(db.Text, nullable=False) # store task/quiz data
+    
+    if activity_type == ActivityType.QUIZ:
+        questions = db.relationship('QuizQuestion', backref='activity', lazy=True, cascade='all, delete-orphan', 
+                                primaryjoin="and_(Activity.id == QuizQuestion.activity_id, Activity.activity_type == 'QUIZ')")
+    elif activity_type == ActivityType.TASK:
+        questions = db.relationship('TaskQuestion', backref='task', lazy=True, cascade='all, delete-orphan', 
+                                primaryjoin="and_(Activity.id == TaskQuestion.activity_id, Activity.activity_type == 'TASK')")
+        
+
+# Quiz Questions
+class QuizQuestion(db.Model):
+    id = db.Column(db.Integer, primary_key=True) # id for identify quiz question
+    quiz_id = db.Column(db.Integer, db.ForeignKey('activity.id'), nullable=False)
+    level = db.Column(db.Integer, nullable=False) # how difficult is the question (easy, medium, hard)
+    active = db.Column(db.Boolean, nullable=False) # determine whether the question is submitted or not
+    score = db.Column(db.Integer, nullable=False) # score for that question
+    question_content = db.Column(db.Text, nullable=False)
+
+# Task Questions
+class TaskQuestion(db.Model):
+    id = db.Column(db.Integer, primary_key=True) # id for identifying task question
+    task_id = db.Column(db.Integer, db.ForeignKey('activity.id'), nullable=False) # id for identifying parent task
+    level = db.Column(db.Integer, nullable=False) # how difficult is the question (easy, medium, hard)
+    active = db.Column(db.Boolean, nullable=False) # determine whether the question is submitted or not
+    score = db.Column(db.Integer, nullable=False) # score for that question
+    question_content = db.Column(db.Text, nullable=False)
+
+# Quiz Answers
+class QuizAnswer(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    quiz_id = db.Column(db.Integer, db.ForeignKey('quiz.id'), nullable=False) # id to identify with parent quiz
```

Comment:
Does foreignKey just restrict the possible values of this column to ones related to the column of another model specified in the string parameter? I'm assuming this due to the foreignKeys above referencing a field of the same name within the activity model/class. If so, then what do the foreignKeys here and on line 126 reference (since there's no quiz or task model)?

If not, then how does foreignKey work, and how do you know what sorts of arguments to pass within it?